src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -664,6 +664,11 @@ bool StorageKafka::streamToViews()
     size_t rows = 0;
     {
         block_io.pipeline.complete(std::move(pipe));
+
+        // we need to read all consumers in parallel (sequential read may lead to situation
+        // when some of consumers are not used, and will break some Kafka consumer invariants)
+        block_io.pipeline.setNumThreads(stream_count);
+
         block_io.pipeline.setProgressCallback([&](const Progress & progress) { rows += progress.read_rows.load(); });
```

Comment:
`rows` should be an atomic variable, otherwise, it may lead to data race, see #36015.